internal/commands/server/k8s_e2e_test.go

```diff
@@ -702,52 +709,76 @@ func TestTCPMeshService(t *testing.T) {
+									Name:      "consul-server-cert",
+									Namespace: &gatewayNamespace,
+								}},
+							},
 						},
-					}},
-				},
-			}
-			err = resources.Create(ctx, gw)
-			require.NoError(t, err)
-			require.Eventually(t, gatewayStatusCheck(ctx, resources, gatewayName, namespace, gatewayReady), 30*time.Second, 1*time.Second, "no gateway found in the allotted time")
-
-			portOne := gateway.PortNumber(serviceOne.Spec.Ports[0].Port)
-			route := &gateway.TCPRoute{
-				ObjectMeta: meta.ObjectMeta{
-					Name:      routeOneName,
-					Namespace: namespace,
-				},
-				Spec: gateway.TCPRouteSpec{
-					CommonRouteSpec: gateway.CommonRouteSpec{
-						ParentRefs: []gateway.ParentRef{{
-							Name: gateway.ObjectName(gatewayName),
-						}},
-					},
-					Rules: []gateway.TCPRouteRule{{
-						BackendRefs: []gateway.BackendRef{{
-							BackendObjectReference: gateway.BackendObjectReference{
-								Name: gateway.ObjectName(serviceOne.Name),
-								Port: &portOne,
+						{
+							Name:     gateway.SectionName(listenerTwoName),
+							Port:     gateway.PortNumber(listenerTwoPort),
+							Protocol: gateway.TCPProtocolType,
+							TLS: &gateway.GatewayTLSConfig{
+								CertificateRefs: []*gateway.SecretObjectReference{{
+									Name:      "consul-server-cert",
+									Namespace: &gatewayNamespace,
+								}},
+								Options: map[gateway.AnnotationKey]gateway.AnnotationValue{
+									"api-gateway.consul.hashicorp.com/tls_min_version":   "TLSv1_1",
+									"api-gateway.consul.hashicorp.com/tls_cipher_suites": "TLS_RSA_WITH_AES_128_CBC_SHA",
+								},
 							},
-						}},
-					}},
+						},
+					},
 				},
 			}
-			err = resources.Create(ctx, route)
+			err = resources.Create(ctx, gw)
 			require.NoError(t, err)
+			require.Eventually(t, gatewayStatusCheck(ctx, resources, gatewayName, namespace, conditionReady), 30*time.Second, 1*time.Second, "no gateway found in the allotted time")
+
+			createTCPRoute(ctx, t, resources, namespace, gatewayName, gateway.SectionName(listenerOneName), routeOneName, serviceOne.Name, gateway.PortNumber(serviceOne.Spec.Ports[0].Port))
+			createTCPRoute(ctx, t, resources, namespace, gatewayName, gateway.SectionName(listenerTwoName), routeTwoName, serviceTwo.Name, gateway.PortNumber(serviceTwo.Spec.Ports[0].Port))
+
+			checkTCPTLSRoute(t, listenerOnePort, &tls.Config{
+				InsecureSkipVerify: true,
+			}, serviceOne.Name, "service not routable in allotted time")
+
+			// Force insecure cipher suite excluded from Consul API Gateway default
+			// cipher suites, but supported by Envoy defaults
+			//
+			// FIXME: why is this test not failing? attempting to connect with only
+			// this cipher suite should be rejected by the Consul API Gateway listener
+			// default config
+			checkTCPTLSRoute(t, listenerOnePort, &tls.Config{
+				InsecureSkipVerify: true,
+				CipherSuites:       []uint16{tls.TLS_RSA_WITH_AES_128_CBC_SHA},
+			}, serviceOne.Name, "service not routable in allotted time")
```

Comment:
Oof, so it looks like I may have been encoding this incorrectly in the Consul > Envoy xDS config generation, I was using a comma-delimited string of cipher suites when it looks like the correct syntax appears to be `"[SUITE_ONE|SUITE_TWO]"`, thus causing my invalid config to be ignored in favor of Envoy's defaults 😭

Reply:
the beauty of what e2e tests catch in this world of multi-service distributed deployments 😬

Reply:
Hmm, maybe that comment from 2018 was using v2 syntax, it does look like a [repeated string](https://www.envoyproxy.io/docs/envoy/v1.20.1/api-v3/extensions/transport_sockets/tls/v3/common.proto#envoy-v3-api-msg-extensions-transport-sockets-tls-v3-tlsparameters) type in v3

Reply:
Oops, this config was being ignored because the TLS client and server negotiated TLS 1.3, which doesn't allow manually configuring cipher suites.